File: Microservices/IoT.House.Automation.Microservices.Arduino/IoT.House.Automation.Microservices.Arduino.Domain/Events/Event.cs
using System;
using IoT.House.Automation.Microservices.Arduino.Domain.Enums;

namespace IoT.House.Automation.Microservices.Arduino.Domain.Events
{
    public class Event
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public EventInputType ExpectedInputType { get; set; }

        public virtual Type Type => this.GetType();


        public virtual bool IsValid()
        {
            return !string.IsNullOrWhiteSpace(Name) && !string.IsNullOrWhiteSpace(Description);
        }
    }
}

File: Microservices/IoT.House.Automation.Microservices.Mobile/domain/src/main/java/iot/house/automation/microservices/mobile/domain/models/arduino/events/Event.java
package iot.house.automation.microservices.mobile.domain.models.arduino.events;

import lombok.Getter;

@Getter
public class Event {
    private String name;
    private String description;
    private String expectedInputType;
    private final String type;

    public Event() {
        type = this.getClass().getSimpleName();
    }
}
